### ISSUE TITLE
Crash when core is polling input during a drivers reinit

### ISSUE THREAD

**Issue by flyinghead** [libretro/RetroArch#12882]
A crash sometimes happens when toggling fullscreen/windowed while the Flycast core is running.
```
#3  0x00005555555dc3d9 in input_state_wrap
    (current_input=0x55555640fa60 <input_x>, data=0x0, joypad=0x0, sec_joypad=0x0, joypad_info=0x7fff81d47310, binds=0x5555564314b0 <rarch_st+124240>, keyboard_mapping_blocked=0x0, port=0x0, device=0x1, idx=0x0, id=0x100) at retroarch.c:481
#4  0x000055555560c3fd in input_state_internal (port=0x0, device=0x1, idx=0x0, id=0x100) at retroarch.c:22775
#5  0x000055555560c848 in input_state (port=0x0, device=0x1, idx=0x0, id=0x100) at retroarch.c:22936
#6  0x00007fff612e9f62 in UpdateInputState (port=0x0) at /home/raph/flycast-option/shell/libretro/libretro.cpp:2349
#7  UpdateInputState(u32) (port=0x0) at /home/raph/flycast-option/shell/libretro/libretro.cpp:2314
#8  0x00007fff612ec93f in UpdateInputState() () at /home/raph/flycast-option/shell/libretro/libretro.cpp:2632
#9  0x00007fff6175329a in maple_DoDma() () at /home/raph/flycast-option/core/hw/maple/maple_if.cpp:150
```

`rarch_st.joypad` is null when the crash happens, presumably because the input driver is being reset.

**Comment by twinaphex** [comment]
Created this branch for now so we can experiment -

https://github.com/libretro/RetroArch/tree/input-robustness-for-threaded-cores

**Comment by twinaphex** [comment]
I think this has been mostly hardened/resolved, but I can move it to 1.9.10 in case it still needs our attention.

**Comment by twinaphex** [comment]
HI @flyinghead , does this issue still happen with the latest master?
